pkg/manifests/common_test.go

```diff
@@ -39,6 +39,27 @@ func TestNamespaceResources(t *testing.T) {
 	}
 }
 
+func TestHasRequiredLabels(t *testing.T) {
+	placeholderPodLabels := map[string]string{"app": "app-name"}
+
+	cases := []struct {
+		Labels    map[string]string
+		ReqLabels []map[string]string
+		Outcome   bool
+	}{
+		{Labels: map[string]string{}, ReqLabels: []map[string]string{GetTopLevelLabels()}, Outcome: false},
+		{Labels: map[string]string{"fake": "fake"}, ReqLabels: []map[string]string{GetTopLevelLabels()}, Outcome: false},
+		{Labels: map[string]string{"app.kubernetes.io/managed-by": "false-operator-name"}, ReqLabels: []map[string]string{GetTopLevelLabels()}, Outcome: false},
+		{Labels: map[string]string{"fakeLabel1": "fakeValue1", "fakeLabel2": "fakeValue2", "fakeLabel3": "fakeValue3", "app.kubernetes.io/managed-by": "aks-app-routing-operator"}, ReqLabels: []map[string]string{GetTopLevelLabels()}, Outcome: true},
+		{Labels: map[string]string{"fakeLabel1": "fakeValue1", "fakeLabel2": "fakeValue2", "fakeLabel3": "fakeValue3", "app.kubernetes.io/managed-by": "aks-app-routing-operator"}, ReqLabels: []map[string]string{GetTopLevelLabels(), placeholderPodLabels}, Outcome: false},
```

Comment:
adding a test case for ReqLabels with at least one key in common would improve testing thoroughness